oligo_designer_toolsuite/database/_reference_database.py

```diff
@@ -1,276 +1,130 @@
-                f"No annotation release defined. Using default release {annotation_release}!"
-            )
-
-        files_source = "NCBI"
-        dir_annotation = os.path.join(dir_output, "annotation")
-        Path(dir_annotation).mkdir(parents=True, exist_ok=True)
-
-        ftp = FTPLoaderNCBI(dir_annotation, species, annotation_release)
-        annotation_file, annotation_release, genome_assembly = ftp.download_files("gtf")
-        sequence_file, _, _ = ftp.download_files("fasta")
-
-        super().__init__(
-            annotation_file,
-            sequence_file,
-            species,
-            genome_assembly,
-            annotation_release,
-            files_source,
-            dir_output,
-        )
-
-
-class EnsemblReferenceDB(CustomReferenceDB):
-    """
-    Class to create reference DB using gtf and fasta files taken from the Ensembl server.
-
-    Sets species, genome_assembly, annotation_release to a predefined value if thay are not given in input.
-    Dowloads the fasta and annotation files from the Ensemble server and stores them in the dir_output folder.
+        """Filter database based on region id and overwrite database with filtered database.
 
-    :param species: species of the files to dowload, defaults to None
-    :type species: str, optional
-    :param annotation_release: annotation_release of the files to dowload, defaults to None
-    :type annotation_release: str, optional
-    :param dir_output: directory where the files are saved, defaults to './output/annotation'
-    :type dir_output: str, optional
-    """
-
-    def __init__(
-        self,
-        species: str = None,
-        annotation_release: str = None,
-        dir_output: str = "output",
-    ):
-        """
-        Constructor
+        :param region_ids: List of region ids.
+        :type region_ids: list
         """
-
-        if species is None:  # change to some standard values for Ensemble
-            species = "human"
-            warnings.warn(f"No species defined. Using default species {species}!")
-
-        if annotation_release is None:
-            annotation_release = "current"
-            warnings.warn(
-                f"No annotation release defined. Using default release {annotation_release}!"
-            )
-
-        files_source = "Ensembl"
-        dir_annotation = os.path.join(dir_output, "annotation")
-
-        Path(dir_annotation).mkdir(parents=True, exist_ok=True)
-        ftp = FtpLoaderEnsembl(dir_annotation, species, annotation_release)
-        annotation_file, annotation_release, genome_assembly = ftp.download_files("gtf")
-        sequence_file, _, _ = ftp.download_files("fasta")
-
-        super().__init__(
-            annotation_file,
-            sequence_file,
-            species,
-            genome_assembly,
-            annotation_release,
-            files_source,
-            dir_output,
-        )
+        database_filtered = []
+        for entry in self.database:
```

Comment:
here the database has to be filled beforehand, amd in case it is empty it does not give any warning. we could directly call the `load_fasta_into_database` function here.